devito/passes/clusters/implicit.py

```diff
@@ -94,8 +91,10 @@ def callback(self, clusters, prefix):
                 processed.append(c)
                 continue
 
-            # Get the dynamic thickness mapper for the given MultiSubDomain
-            mapper, dims = lower_msd(d.msd, c)
+            # Get all MultiSubDimensions in the cluster and get the dynamic thickness
+            # mapper for the associated MultiSubDomain
+            mapper, dims = lower_msd({msdim(i.dim) for i in c.ispace[idx:]} - {None})
```

Comment:
I'm not sure I understand this change. If you are visiting `d`, why do you need its msd as well as any other nested msds ? 

I don't think this is particularly clean, but besides that, I'm failing to understand why

Reply:
`lower_msd` has been completely replaced. Now stands for "lower multisubdimension" and takes all the multisubdimensions present in the cluster, rather than the multisubdomain attached to the outermost dimension (all references to which have been dropped at this point).